mantidproject/mantid
# New algorithm to make ins file for shelx - SaveINS

**Description of work.**

New algorithm to make ins file for shelx - see algorithm doc file for details.



**To test:**

(1) Run this script
```
from mantid.simpleapi import *

LoadEmptyInstrument(InstrumentName='SXD', OutputWorkspace='SXD')
SetUB(Workspace='SXD', a=7.6508, b=13.2431, c=11.6243, alpha=90, beta=104.1183, gamma=90)
ndensity = 4/mtd['SXD'].sample().getOrientedLattice().volume()
SetSample(InputWorkspace='SXD', Material={'ChemicalFormula': 'C12 H9 N3 O2 S1', 'SampleNumberDensity': ndensity})
SaveINS(InputWorkspace='SXD', Filename=r"C:\Users\xhg73778\Desktop\test.ins", Spacegroup='P 1 21/n 1')
SaveINS(InputWorkspace='SXD', Filename=r"C:\Users\xhg73778\Desktop\test_isotope.ins", Spacegroup='P 1 21/n 1', UseNaturalIsotopicAbundances=False)
```
(2) Check the files are saved
- The `test,ins` file should have a single `SFAC` line (`SFAC C H N O S`)
- The `test_isotope.ins` should print out more info (scattering lengths etc.) for each atom.




Fixes #34275 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
